lib/ai/model/output/messages.go

```diff
@@ -26,6 +28,16 @@ type StreamingMessage struct {
 	Parts <-chan string
 }
 
+// String implements the Stringer interface. It waits until the message stream
+// is over and returns the full message.
+func (msg *StreamingMessage) String() string {
+	sb := strings.Builder{}
+	for part := range msg.Parts {
```

Comment:
This sounds hacky. As the `Parts` is blocking you need to create a `StreamingMessage` and print it, otherwise everything will block. Moreover, you cannot print it twice. 
I think this should be renamed to show the intent. If you want to be able to print it, I'd cache the result string in the memory., so `String()` can be called multiple times, but I don't think that this is really needed.